worlds/dark_souls_3/Items.py

```diff
@@ -12,6 +15,22 @@ def get_name_to_id() -> dict:
 
         output = {}
         for i, table in enumerate(item_tables):
+            if table.__len__() > table_offset:
+                print("An item table contains {} entries, that is more than {} entries ({})".format(table.__len__(), table_offset, i))
+                sys.exit(1)
             output.update({name: id for id, name in enumerate(table, base_id + (table_offset * i))})
 
         return output
+
+    @staticmethod
+    def is_dlc_item(name) -> bool:
+        return name in dlc_shields_table \
+            or name in dlc_weapons_upgrade_10_table \
+            or name in dlc_weapons_upgrade_5_table \
+            or name in dlc_goods_table \
+            or name in dlc_spells_table \
+            or name in dlc_armor_table \
+            or name in dlc_ring_table \
+            or name in dlc_misc_table
```

Comment:
for efficiency sake I'd suggest building a set from all the dlc lists once ahead of item and then check if the name is in that set.

Reply:
Done